doc/source/release/1.23.0-notes.rst

```diff
@@ -4,42 +4,449 @@
+  same meaning as ``axis=None``.  This is deprecated and ``axis=None``
+  must be used instead.
 
-Future Changes
-==============
+  (`gh-20920 <https://github.com/numpy/numpy/pull/20920>`__)
+
+* The hook function ``PyDataMem_SetEventHook`` has been deprecated and the
+  demonstration of its use in tool/allocation_tracking has been removed.  The
+  ability to track allocations is now built-in to python via ``tracemalloc``.
+
+  (`gh-20394 <https://github.com/numpy/numpy/pull/20394>`__)
+
+* ``numpy.distutils`` has been deprecated, as a result of ``distutils`` itself
+  being deprecated. It will not be present in NumPy for Python >= 3.12, and
+  will be removed completely 2 years after the release of Python 3.12 For more
+  details, see :ref:`distutils-status-migration`.
+
+  (`gh-20875 <https://github.com/numpy/numpy/pull/20875>`__)
 
 
 Expired deprecations
 ====================
 
+* The ``NpzFile.iteritems()`` and ``NpzFile.iterkeys()`` methods have been
+  removed as part of the continued removal of Python 2 compatibility. This
+  concludes the deprecation from 1.15.
 
-Compatibility notes
-===================
+  (`gh-16830 <https://github.com/numpy/numpy/pull/16830>`__)
 
+* The ``alen`` and ``asscalar`` functions have been removed.
 
-C API changes
-=============
+  (`gh-20414 <https://github.com/numpy/numpy/pull/20414>`__)
+
+* The ``UPDATEIFCOPY`` array flag has been removed together with the enum
+  ``NPY_ARRAY_UPDATEIFCOPY``. The associated (and deprecated)
+  ``PyArray_XDECREF_ERR`` was also removed. These were all deprecated in 1.14. They
+  are replaced by ``WRITEBACKIFCOPY``, that requires calling
+  ``PyArray_ResoveWritebackIfCopy`` before the array is deallocated.
+
+  (`gh-20589 <https://github.com/numpy/numpy/pull/20589>`__)
+
+* Exceptions will be raised during array-like creation.  When an object raised
+  an exception during access of the special attributes ``__array__`` or
+  ``__array_interface__``, this exception was usually ignored.  This behaviour
+  was deprecated in 1.21, and the exception will now be raised.
+
+  (`gh-20835 <https://github.com/numpy/numpy/pull/20835>`__)
+
+* Multidimensional indexing with non-tuple values is not allowed.  Previously,
+  code such as ``arr[ind]`` where ``ind = [[0, 1], [0, 1]]`` produced a
+  ``FutureWarning`` and was interpreted as a multidimensional index (i.e.,
+  ``arr[tuple(ind)]``). Now this example is treated like an array index over a
+  single dimension (``arr[array(ind)]``).  Multidimensional indexing with
+  anything but a tuple was deprecated in NumPy 1.15.
+
+  (`gh-21029 <https://github.com/numpy/numpy/pull/21029>`__)
+
+* Changing to a dtype of different size in F-contiguous arrays is no longer
+  permitted. Deprecated since Numpy 1.11.0. See below for an extended
+  explanation of the effects of this change.
+
+  (`gh-20722 <https://github.com/numpy/numpy/pull/20722>`__)
+
+
+The behavior change associated with disallowing the dtype size changes in
```

Comment:
Do we want to omit this? Maybe move it somewhere else?

Reply:
:+1: on simply removing the whole block and only keeping your nice bullet point.